src/NuGet.Services.Validation.Orchestrator/ContentScan/ContentScanValidator.cs

```diff
@@ -0,0 +1,132 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NuGet.Jobs.Validation;
+using NuGet.Jobs.Validation.ContentScan;
+using NuGet.Jobs.Validation.Storage;
+using NuGet.Services.Entities;
+using NuGetGallery;
+
+namespace NuGet.Services.Validation.Orchestrator.ContentScan
+{
+    [ValidatorName(ValidatorName.ContentScanValidator)]
+    public class ContentScanValidator : IValidator
+    {
+        private readonly IValidationEntitiesContext _validationContext;
+        private readonly IValidatorStateService _validatorStateService;
+        private readonly IContentScanEnqueuer _contentScanEnqueuer;
+        private readonly ContentScanConfiguration _configuration;
+        private readonly ILogger<ContentScanValidator> _logger;
+
+        public ContentScanValidator(
+            IValidationEntitiesContext validationContext,
+            IValidatorStateService validatorStateService,
+            IContentScanEnqueuer contentScanEnqueuer,
+            IOptionsSnapshot<ContentScanConfiguration> configurationAccessor,
+            ILogger<ContentScanValidator> logger)
+        {
+            _validationContext = validationContext ?? throw new ArgumentNullException(nameof(validationContext));
+            _validatorStateService = validatorStateService ?? throw new ArgumentNullException(nameof(validatorStateService));
+            _contentScanEnqueuer = contentScanEnqueuer ?? throw new ArgumentNullException(nameof(contentScanEnqueuer));
+
+            if (configurationAccessor == null)
+            {
+                throw new ArgumentNullException(nameof(configurationAccessor));
+            }
+            if (configurationAccessor.Value == null)
+            {
+                throw new ArgumentException($"{nameof(configurationAccessor.Value)} property is null", nameof(configurationAccessor));
+            }
+            _configuration = configurationAccessor.Value;
+
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            configurationAccessor = configurationAccessor ?? throw new ArgumentNullException(nameof(configurationAccessor));
+
+            if (configurationAccessor.Value == null)
+            {
+                throw new ArgumentException($"{nameof(configurationAccessor.Value)} property is null", nameof(configurationAccessor));
+            }
+
+            _configuration = configurationAccessor.Value;
+        }
+
+        public async Task<IValidationResponse> StartAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var validatorStatus = await GetValidatorStatusAsync(request);
+
+            if (validatorStatus.State != ValidationStatus.NotStarted)
+            {
+                _logger.LogWarning(
+                    "Scan and Sign validation with validation Id {ValidationStepId} has already started.",
```

Comment:
"Scan and sign" is the malware scanning step, this should be the "content scan" step:

```suggestion
                    "Content scan with validation Id {ValidationStepId} has already started.",
```